src/components/PageSettingsModal.jsx

```diff
@@ -53,32 +57,41 @@ const PageSettingsModal = ({
       permalink: setPermalink,
     }
 
-    const { data } = useQuery(
+    const {} = useQuery(
       'page',
-      () => {},  // for page settings editing 
+      async () => await getPage(pageType, siteName, folderName, originalPageName),
       { 
         enabled: !isNewPage,
-        onSuccess: () => {
-          setTitle(data.name)
-          setPermalink(data.permalink)
-          setSha(data.sha)
-          setMdBody(data.mdBody)
+        retry: false,
+        onSuccess: ({ content, sha }) => {
+          const { frontMatter, mdBody } = frontMatterParser(content)
+          setTitle(deslugifyPage(originalPageName))
+          setPermalink(frontMatter.permalink)
+          setOriginalPermalink(frontMatter.permalink)
+          setSha(sha)
+          setMdBody(mdBody)
         }, 
-        onError: () => errorToast(`The page data could not be retrieved. Please try again. ${DEFAULT_ERROR_TOAST_MSG}`)
+        onError: () => { 
+          setSelectedPage('')
+          setIsPageSettingsActive(false)
+          errorToast(`The page data could not be retrieved. Please try again. ${DEFAULT_ERROR_TOAST_MSG}`)
+        }
       }
     )
 
     const { mutateAsync: saveHandler } = useMutation(
-      async () => { 
-        const fileInfo = { siteName, title, permalink, mdBody, folderName, subfolderName, pageType }
+      async () => {
+        if (originalPageName === generatePageFileName(title) && originalPermalink === permalink) return 
+        const fileInfo = { siteName, title, permalink, mdBody, folderName, subfolderName, isNewPage, pageType, originalPageName }
         const { endpointUrl, content, redirectUrl } = generatePageContent(fileInfo)
-        await createPage(endpointUrl, content)
+        if (isNewPage) await createPage(endpointUrl, content)
+        if (!isNewPage) await updatePage(endpointUrl, content, sha)
         return redirectUrl
       },
       { 
-        onSettled: () => setIsPageSettingsActive(false),
-        onSuccess: (redirectUrl) => setRedirectToPage(redirectUrl),  
-        onError: () => errorToast(`A new page could not be created. Please try again. ${DEFAULT_ERROR_TOAST_MSG}`)
+        onSettled: () => {setSelectedPage(''); setIsPageSettingsActive(false)},
+        onSuccess: (redirectUrl) => redirectUrl ? setRedirectToPage(redirectUrl) : window.location.reload(),
```

Comment:
I misread this: `redirectUrl` is the `data` that's passed back from the `useMutation` callback, so there're no problems here

Reply:
discussed offline - the issue wasn't with `redirectUrl` as there are two functions above with different scope. However, it wasn't clear where the scope of the mutation function ended which made this code chunk confusing. This will be refactored in a separate PR.